cg/cli/deliver/base.py

```diff
@@ -85,3 +87,57 @@ def rsync(context: CGConfig, ticket_id: int, dry_run: bool):
     """
     rsync_api = RsyncAPI(config=context)
     rsync_api.run_rsync_command(ticket_id=ticket_id, dry_run=dry_run)
+
+
+@deliver.command(name="concatenate")
+@click.argument("ticket_id", type=int, required=True)
+@click.pass_context
+def concatenate(context: click.Context, ticket_id: int):
+    """The fastq files in the folder generated using "cg deliver analysis"
+    will be concatenated into one forward and one reverse fastq file.
+    """
+    cg_context: CGConfig = context.obj
+    deliver_ticket_api = DeliverTicketAPI(config=cg_context)
+    deliver_ticket_api.concatenate(ticket_id=ticket_id)
+
+
+@deliver.command(name="ticket")
+@click.option(
+    "-t",
+    "--ticket-id",
+    type=int,
+    help="Deliver and rsync the files for ALL cases connected to a ticket",
+    required=True,
+)
+@click.option("-d", "--delivery-type", type=click.Choice(PIPELINE_ANALYSIS_OPTIONS), required=True)
+@click.option("--dry-run", is_flag=True)
+@click.pass_context
+def deliver_ticket(
+    context: click.Context,
+    ticket_id: int,
+    delivery_type: str,
+    dry_run: bool,
+):
+    """Will first collect hard links in the customer inbox then
+    concatenate fastq files if needed and finally send the folder
+    from customer inbox hasta to the customer inbox on caesar.
+    """
+    cg_context: CGConfig = context.obj
+    deliver_ticket_api = DeliverTicketAPI(config=cg_context)
+    is_upload_needed = deliver_ticket_api.check_if_upload_is_needed(ticket_id=ticket_id)
```

Comment:
could yo hide more logic inside the DeliverTicketAPI?
Try to refactor so that the checks for upload needed, analysis type, concatenation needed, etc, can be handled within the class, and the conditions are not resolved both in the class and in the cli. Error handling and invocation of other cli commands should stay in the cli part of code

Reply:
Agree, please avoid logic in the CLI as much as possible